### ISSUE TITLE
Single result page should use header with navigation links

### ISSUE THREAD

**Issue by dhruvkb** [WordPress/openverse#463]
## Problem

Currently the single result page shows a header with the following items
- Openverse logo (link to homepage)
- Search bar
- Media type selection dropdown
- Filter sidebar toggle

The last two have no meaning on the single result page as you cannot change the media type of the result you are seeing, nor can you filter it down any further.

## Description

According to me this can be replaced with the other header that contains navigation links.
- The search bar will still be accessible, although with an extra click of going to the homepage first.
- The new list of links will be better for someone has does not know about Openverse but was sent a link to a media on it.

## Alternatives

We just hide the two meaningless controls and continue to use the same header as now.

## Additional context

This needs design input.

**Comment by zackkrida** [comment]
I understand wanting to reinforce the search action, but it does feel like a header where ~30% is unusable might not be the best use of space on the page. I think after viewing a single result the most likely actions a user would want to take are:

- Go back to the results
- Make a new search

That supports showing the search bar _or_ expecting users to use the "back to search results" button. If we assume that button covers the two primary use cases, we could then use the header to show pages like @dhruvkb suggested.

I am open to either approach, though, and just sharing ideas.

**Comment by panchovm** [comment]
Sorry but I am not clear about your point. I understand you vote for showing the header with internal links. Correct me if I'm wrong.

I agree that those two actions are the most likely ones, as they both belong to the search flow. Therefore, showing the search bar and the "back to search" action in this page at the same time solves the need. Then, how the content header is better for that purpose?

The header would not show the search term typed and the only way of searching is going back to the homepage. Changing the main nav while searching breaks the flow.

I checked this flow on other similar sites, but most display the single result in a modal. However, Youtube is a good example of going back and forth between results and the header remains the same.

**Comment by obulat** [comment]
I do often search from the single result page, so personally, I would prefer for the search bar to stay. But I'm pretty sure my behavior is different from what our users normally do, and there are convincing arguments for removing the search bar from the single results page.

Disabling the search type and filter buttons on the single result header make the search experience really different from what it is on the search page: you can only change the search term, and don't have the control over the other options. This fact makes me lean towards @dhruvkb 's suggestion of using the content header (with the page links) on the single result page. This will mean that if you load a page directly, you would need to go to the homepage to search (which makes the discovery a bit worse, but how often do users land on a page directly?)

@panchovm , on your Youtube example: they do display the search bar on every page, but they only display the filters on the search page. For us, the analogy would be to remove the search type and filter buttons completely instead of making them disabled, I guess. It wouldn't be a complete analogy because the Youtube filters are not in the header anyways...

When we have single results in a modal, this will be simplified :)

**Comment by panchovm** [comment]
I made several prototypes to test how the header looks in the single result page. Here are the options we are handling.

## 1. `Header content` with switcher and filters disabled

https://user-images.githubusercontent.com/895819/225652348-358eb8cf-af74-4107-a0ce-f69869eddca7.mp4

## 2. `Header internal` sticky

https://user-images.githubusercontent.com/895819/225652383-5b70b2da-78cd-4e04-a227-b98bb83359d4.mp4

## 3. `Header internal` non-sticky

https://user-images.githubusercontent.com/895819/225652418-81a2b958-2ac3-4994-b64f-f96adee02ee2.mp4

---

I personally like option 1 as it keeps the search bar, but it is also true that disabling the actions feel odd. In that case, option 3 is more friendly if we disable the sticky behavior.

It also works for landing in single result pages when coming from openverse links but without the "back to results" area and a clear search bar.

**Comment by obulat** [comment]
I like option 1 for the ability to search from the single result page, especially for the users who start from the single result. What I don't like about it is that the filters/search type buttons are disabled and their status might be arbitrary. For an image of a bird, it could say "Images"/"All Content". If you open the same image from the search page and directly, the state of the buttons will be different.

For option 3, I like the simplicity of not having to decide what the button status should be, but hate the fact that to search again, if you landed on the page directly, you would have to go to the homepage.
I was going to suggest that maybe we should have a "Go to search page" link on the place of "Back to results" when the user lands on that page directly, but then realized that we can't have a search page without a search term - it redirects to the homepage.

I wish we knew how many users search from the single result page's header to make a more informed decision.

**Comment by panchovm** [comment]
> For an image of a bird, it could say "Images"/"All Content". If you open the same image from the search page and directly, the state of the buttons will be different.

I tried this flow, and the switcher label shows the content type results I'm coming from. The label is consistent.

I am 100% convinced that removing the search bar harms the browsing flow. And I bet that's why other big players like Unsplash, Pexels, and Rawpixel have the single result in a modal. But for the purpose of this transitional case, until having the modal integrated, shifting to `header internal` from `header content` seems the easiest solution to solve the flow inconsistency.

While reviewing this case, I also explored other solutions. But all of them require modifying either the content or internal version of the header.

**Comment by obulat** [comment]
> I tried this flow, and the switcher label shows the content type results I'm coming from. The label is consistent.

Try searching for "bird" in images: https://openverse.org/search/image?q=birds, and open one of the results. You'll see that the content switcher will show "Images". Refresh the page - the switcher will show "All content".

I hope we can get to showing the results in a modal soon.

**Comment by zackkrida** [comment]
We have [decided](https://wordpress.slack.com/archives/C02012JB00N/p1679413973067849) to move forward with Option 1, the disabled state, and this issue is now ready for work.

CC @WordPress/openverse-frontend.

**Comment by obulat** [comment]
> Will their new search inherit the filters and content type of their previous search, or be a new, "blank" search with their input search term?

I was thinking that we would leave the previous state (search type and filters) if we have any, but if we don't (as on the server-rendered single result page), we would use the defaults (no filters, All content).

**Comment by zackkrida** [comment]
> The component was conceived with the modal navigation, side by side.

This is very important to remember 👍 A question, @panchovm. When we implement single result modals, the results will _not_ be in a modal when accessed directly, via url, correct? So we still need to figure out a solution for this issue even in the case of using modals.

**Comment by panchovm** [comment]
Correct. In that case we should show the `header internal`. Searching will require clicking on the openverse logo and going to the homepage.
